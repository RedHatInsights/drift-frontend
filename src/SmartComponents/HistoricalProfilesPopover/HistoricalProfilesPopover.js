import React, { useState, useEffect } from 'react';
import { Badge, Button, Popover } from '@patternfly/react-core';
import { useDispatch, useSelector } from 'react-redux';
import { ExclamationCircleIcon, HistoryIcon, UndoIcon } from '@patternfly/react-icons';
import PropTypes from 'prop-types';
import { Skeleton, SkeletonSize } from '@redhat-cloud-services/frontend-components';

import systemsTableActions from '../SystemsTable/actions';
import HistoricalProfilesCheckbox from './HistoricalProfilesCheckbox/HistoricalProfilesCheckbox';
import api from '../../api';
import EmptyStateDisplay from '../EmptyStateDisplay/EmptyStateDisplay';
import HistoricalProfilesRadio from './HistoricalProfilesRadio/HistoricalProfilesRadio';
import addSystemModalActions from '../AddSystemModal/redux/actions';

const HistoricalProfilesPopover = ({
    badgeCount,
    fetchCompare,
    hasBadge,
    hasCompareButton,
    hasMultiSelect,
    referenceId,
    selectedBaselineIds,
    selectHistoricProfiles,
    selectSystem,
    system,
    systemIds,
    systemName
}) => {
    const dispatch = useDispatch();
    const selectedHSPIds = useSelector(({ historicProfilesState }) => historicProfilesState.selectedHSPIds);
    const [ historicalData, setHistoricalData ] = useState();
    const [ isLoading, setIsLoading ] = useState(true);
    const [ localBadgeCount, setLocalBadgeCount ] = useState(badgeCount ? badgeCount : 0);
    const [ error, setError ] = useState();

    const hasHistoricalData = () => {
        return historicalData && historicalData.profiles.length > 0;
    };

    const onSelect = async (checked, profile) => {
        let newSelectedHSPIds = [ ...selectedHSPIds ];

        if (profile.captured_date === 'Latest') {
            await selectSystem(profile.id, !checked);
        } else {
            if (newSelectedHSPIds.includes(profile.id)) {
                newSelectedHSPIds = newSelectedHSPIds.filter(hsp => hsp !== profile.id);
            } else {
                newSelectedHSPIds.push(profile.id);
            }

            await selectHistoricProfiles(newSelectedHSPIds);
        }

        dispatch(addSystemModalActions.handleHSPSelection(profile));
    };

    const onSingleSelect = async (profile) => {
        if (profile.captured_date === 'Latest') {
            await selectSystem(profile.id, true);
        }

        systemsTableActions.selectSingleHSP(profile);
    };

    const updateBadgeCount = () => {
        setLocalBadgeCount(historicalData?.profiles.filter((hsp) => {
            return selectedHSPIds.includes(hsp.id);
        }).length);
    };

    /*eslint-disable camelcase*/
    const fetchData = async () => {
        setIsLoading(true);
        let fetchedData = await api.fetchHistoricalData(system.system_id ? system.system_id : system.id);
        setIsLoading(false);

        fetchedData.profiles?.forEach(function(profile) {
            profile.system_name = systemName;
        });

        if (fetchedData.status) {
            setError({ status: fetchedData.status, message: fetchedData.data.message });
        } else {
            fetchedData.profiles.shift();
            setHistoricalData(fetchedData);
        }
    };
    /*eslint-enable camelcase*/

    const createDropdownArray = () => {
        let dropdownItems = [];

        if (isLoading) {
            for (let i = 0; i < 3; i += 1) {
                dropdownItems.push(
                    <React.Fragment>
                        <Skeleton
                            size={ SkeletonSize.sm }
                            key={ 'skeleton-row-' + i }
                        />
                        <br className='hsp-dropdown-loading' />
                    </React.Fragment>
                );
            }
        } else {
            if (hasHistoricalData()) {
                historicalData.profiles.forEach(function(profile, index) {
                    dropdownItems.push(
                        <div className={ index > 0 ? 'sm-padding-top' : null }>
                            { hasMultiSelect
                                ? <HistoricalProfilesCheckbox
                                    profile={ profile }
                                    updateBadgeCount={ updateBadgeCount }
                                    onSelect={ onSelect }
                                    selectedHSPIds={ selectedHSPIds }
                                />
                                : <HistoricalProfilesRadio
                                    profile={ profile }
                                    onSingleSelect={ onSingleSelect }
                                    selectedHSPIds={ selectedHSPIds }
                                />
                            }
                        </div>
                    );
                });
            } else if (error) {
                dropdownItems.push(
                    <EmptyStateDisplay
                        icon={ ExclamationCircleIcon }
                        isSmall={ true }
                        color='#c9190b'
                        title={ 'Cannot get historical check-ins' }
                        error={ error.status + ': ' + error.message }
                        button={ <a onClick={ () => fetchData() }>
                            <UndoIcon className='reload-button' />
                                Retry
                        </a> }
                    />
                );
            } else {
                dropdownItems.push(
                    <div>
                        There are no historical profiles to display.
                    </div>
                );
            }
        }

        return dropdownItems;
    };

    const [ isVisible, setIsVisible ] = useState(false);
    const [ dropDownArray, setDropDownArray ] = useState(createDropdownArray());

    const runFetchCompare = () => {
        fetchCompare(systemIds, selectedBaselineIds, selectedHSPIds, referenceId);
    };

    const onToggle = () => {
        if (isVisible === false) {
            fetchData();
        }

        setIsVisible(!isVisible);
    };

    const renderBadge = () => {
        if (localBadgeCount > 0) {
            return <Badge key={ 1 }>{ localBadgeCount }</Badge>;
        } else {
            return null;
        }
    };

    useEffect(() => {
        updateBadgeCount();
        setDropDownArray(createDropdownArray());
    }, [ selectedHSPIds ]);

    useEffect(() => {
        setDropDownArray(createDropdownArray());
    }, [ historicalData, error ]);

    let id = system?.system_id ? system?.system_id : system?.id;

    return (
        <React.Fragment>
            <span
                className='hsp-icon-padding'
                data-ouia-component-id={ 'hsp-popover-toggle-' + id  }
                data-ouia-component-type='PF4/Button' >
                <Popover
                    id={ 'hsp-popover-' + id }
                    isVisible={ isVisible }
                    shouldClose={ () => onToggle() }
                    headerContent={ <div>Historical profiles for this system</div> }
                    bodyContent={ <div style={{ maxHeight: '350px', overflowY: 'scroll' }}>
                        { dropDownArray }
                    </div> }
                    footerContent={ hasCompareButton
                        ? <Button
                            variant='primary'
                            ouiaId="hsp-popover-compare"
                            isDisabled={ !hasHistoricalData() }
                            onClick={ () => runFetchCompare() }>
                            Compare
                        </Button>
                        : null }
                >
                    <HistoryIcon className='hsp-dropdown-icon' onClick={ () => onToggle() } />
                </Popover>
            </span>
            { hasBadge ? renderBadge() : null }
        </React.Fragment>
    );
};

HistoricalProfilesPopover.propTypes = {
    badgeCount: PropTypes.number,
    fetchCompare: PropTypes.func,
    handleHSPSelection: PropTypes.func,
    hasBadge: PropTypes.bool,
    hasCompareButton: PropTypes.bool,
    hasMultiSelect: PropTypes.bool,
    referenceId: PropTypes.string,
    selectedBaselineIds: PropTypes.array,
    selectedHSPIds: PropTypes.array,
    selectHistoricProfiles: PropTypes.func,
    selectSingleHSP: PropTypes.func,
    selectSystem: PropTypes.func,
    system: PropTypes.object,
    systemIds: PropTypes.array,
    systemName: PropTypes.string
};

export default HistoricalProfilesPopover;
