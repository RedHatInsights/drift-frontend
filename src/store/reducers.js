import React from 'react';
import { mergeArraysByKey } from '@redhat-cloud-services/frontend-components-utilities/files/helpers';
import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/files/ReducerRegistry';
import HistoricalProfilesPopover from '../SmartComponents/HistoricalProfilesPopover/HistoricalProfilesPopover';
import SystemKebab from '../SmartComponents/BaselinesPage/EditBaselinePage/SystemNotification/SystemKebab/SystemKebab';

import types from '../SmartComponents/modules/types';
import helpers from '../SmartComponents/helpers';

function selectedReducer(
    INVENTORY_ACTIONS, baselineId, createBaselineModal, historicalProfiles, hasMultiSelect, hasHistoricalDropdown,
    deselectHistoricalProfiles, isAddSystemNotifications, selectHistoricProfiles, systemNotificationIds, selectSystemsToAdd,
    deleteNotifications
) {
    let newColumns;
    let systemIds;

    return applyReducerHash({
        [INVENTORY_ACTIONS.LOAD_ENTITIES_FULFILLED]: (state, action) => {
            newColumns = [ ...state.columns ];

            for (let i = 0; i < action.payload.results.length; i += 1) {
                if (state.selectedSystemIds.includes(action.payload.results[i].id)) {
                    action.payload.results[i].selected = true;
                }
            }

            let rows = mergeArraysByKey([ action.payload.results, state.rows ]);

            if (historicalProfiles !== undefined) {
                /*eslint-disable camelcase*/
                if (!newColumns.find(({ key }) => key === 'historical_profiles') && hasHistoricalDropdown) {
                    newColumns.push({
                        key: 'historical_profiles',
                        title: 'Historical profiles',
                        props: {
                            isStatic: true,
                            width: 10
                        }
                    });
                }

                if (newColumns.find(({ key }) => key === 'system_notification')) {
                    newColumns.splice(newColumns.findIndex(({ key }) => key === 'system_notification'), 1);
                }

                rows.forEach(function(row) {
                    let badgeCount = 0;
                    let systemInfo = {
                        id: row.id,
                        last_updated: row.updated
                    };

                    historicalProfiles.forEach(function(profile) {
                        if (profile.system_id === row.id) {
                            badgeCount += 1;
                        }
                    });

                    row.historical_profiles = <React.Fragment>
                        <HistoricalProfilesPopover
                            system={ systemInfo }
                            hasBadge={ true }
                            badgeCount={ badgeCount }
                            hasMultiSelect={ hasMultiSelect }
                            selectHistoricProfiles={ selectHistoricProfiles }
                        />
                    </React.Fragment>;
                });
            }

            if (isAddSystemNotifications) {
                rows.forEach(function(row) {
                    if (systemNotificationIds.includes(row.id)) {
                        row.selected = true;
                        row.disableSelection = true;
                    }
                });
            }

            if (baselineId && !isAddSystemNotifications) {
                if (!newColumns.find(({ key }) => key === 'system_notification')) {
                    newColumns.push({
                        key: 'system_notification',
                        title: '',
                        props: {
                            isStatic: true
                        }
                    });
                }

                if (newColumns.find(({ key }) => key === 'historical_profiles')) {
                    newColumns.splice(newColumns.findIndex(({ key }) => key === 'historical_profiles'), 1);
                }

                rows.forEach(function(row) {
                    systemIds = [ row.id ];

                    row.system_notification = <React.Fragment>
                        <SystemKebab
                            systemIds={ systemIds }
                            systemName={ row.display_name }
                            deleteNotifications={ deleteNotifications }
                        />
                    </React.Fragment>;
                });
            }
            /*eslint-enable camelcase*/

            /* Hide link on systems table */
            newColumns.forEach(function(column, index) {
                if (!column.props) {
                    column.props = {};
                }

                if (hasHistoricalDropdown) {
                    if (column.key === 'display_name' || column.key === 'display_selected_hsp') {
                        column.props.width = 20;
                        if (newColumns.find(({ key }) => key === 'display_selected_hsp')) {
                            let displaySelectedHSP = newColumns.splice(newColumns.findIndex(({ key }) => key === 'display_selected_hsp'), 1);
                            newColumns.splice(index, 1, displaySelectedHSP[0]);
                        }
                    } else {
                        column.props.width = 10;
                    }
                } else {
                    column.props.width = null;
                }

                if (column.composed) {
                    delete column.composed;
                }
            });

            return {
                ...state,
                columns: newColumns,
                rows: state.selectedHSP
                    ? helpers.buildSystemsTableWithSelectedHSP(rows, state.selectedHSP, deselectHistoricalProfiles)
                    : rows
            };
        },
        [types.DRIFT_CLEAR_ALL_FILTERS]: (state) => ({
            ...state,
            activeFilters: undefined
        }),
        [types.UPDATE_COLUMNS]: (state, action) => {
            let nameColumn = {
                key: action.payload,
                title: 'Name',
                props: { width: 20 }
            };
            newColumns = [ ...state.columns ];
            newColumns.shift();
            newColumns.unshift(nameColumn);

            return {
                ...state,
                columns: newColumns
            };
        },
        [types.SELECT_ENTITY]: (state, action) => {
            let id = action.payload.id;
            let selected = action.payload.selected;
            let { selectedSystemIds } = state;
            let newRows = [];

            if (id === 0) {
                if (createBaselineModal) {
                    newRows = state.rows.map(function (oneRow) {
                        oneRow.selected = false;
                        return oneRow;
                    });

                    selectedSystemIds = [];
                } else {
                    if (selected) {
                        let ids = state.rows.map(function (item) {
                            return item.id;
                        });
                        selectedSystemIds = [ ...new Set(selectedSystemIds.concat(ids)) ];
                    } else {
                        if (action.payload.bulk) {
                            selectedSystemIds = [];
                        } else {
                            for (let i = 0; i < state.rows.length; i += 1) {
                                selectedSystemIds = selectedSystemIds.filter(item => item !== state.rows[i].id);
                            }
                        }
                    }
                }
            } else if (selected && id !== null) {
                if (createBaselineModal) {
                    newRows = state.rows.map(function (oneRow) {
                        if (oneRow.id === id) {
                            oneRow.selected = true;
                        } else {
                            oneRow.selected = false;
                        }

                        return oneRow;
                    });

                    selectedSystemIds.pop();
                }

                selectedSystemIds.push(id);
            } else {
                selectedSystemIds = selectedSystemIds.filter(item => item !== id);
            }

            if (isAddSystemNotifications) {
                selectSystemsToAdd(selectedSystemIds);
            }

            return {
                ...state,
                selectedSystemIds,
                rows: newRows.length > 0 ? newRows : state.rows,
                columns: newColumns
            };
        },
        [types.SET_SELECTED_SYSTEM_IDS]: (state, action) => {
            return {
                ...state,
                selectedSystemIds: action.payload.selectedSystemIds
            };
        },
        [types.SELECT_SINGLE_HSP]: (state, action) => {
            return {
                ...state,
                rows: helpers.buildSystemsTableWithSelectedHSP([ ...state.rows ], action.payload, deselectHistoricalProfiles),
                columns: newColumns,
                selectedSystemIds: [],
                selectedHSP: action.payload
            };
        }
    });
}

export default selectedReducer;
