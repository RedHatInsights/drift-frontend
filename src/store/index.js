import promiseMiddleware from 'redux-promise-middleware';
import { getRegistry } from '@redhat-cloud-services/frontend-components-utilities/files/Registry';

import { compareReducer, globalFilterReducer } from '../SmartComponents/modules/reducers';
import { addSystemModalReducer } from '../SmartComponents/AddSystemModal/redux/addSystemModalReducer';
import { baselinesTableRootReducer } from '../SmartComponents/BaselinesTable/redux';
import { filterDropdownReducer } from '../SmartComponents/DriftPage/FilterDropDown/redux/filterDropdownReducer';
import { createBaselineModalReducer } from '../SmartComponents/BaselinesPage/CreateBaselineModal/redux/reducers';
import { editBaselineReducer } from '../SmartComponents/BaselinesPage/EditBaselinePage/redux/reducers';
import { historicProfilesReducer } from '../SmartComponents/HistoricalProfilesPopover/redux/reducers';
import { systemNotificationsReducer } from '../SmartComponents/BaselinesPage/EditBaselinePage/SystemNotification/redux/reducer';

import MiddlewareListener from '@redhat-cloud-services/frontend-components-utilities/files/MiddlewareListener';
import { notifications } from '@redhat-cloud-services/frontend-components-notifications';
export { default as reducers } from './reducers';

let registry;
let middlewareListener;

export const createMiddlewareListener = () => {
    middlewareListener = new MiddlewareListener();
    return middlewareListener;
};

export function init (...middleware) {
    createMiddlewareListener();

    registry = getRegistry({}, [
        promiseMiddleware(),
        middlewareListener.getMiddleware(),
        ...middleware.filter(item => typeof item !== 'undefined')
    ]);

    registry.register({
        compareState: compareReducer,
        addSystemModalState: addSystemModalReducer,
        baselinesTableState: baselinesTableRootReducer,
        filterDropdownOpened: filterDropdownReducer,
        createBaselineModalState: createBaselineModalReducer,
        editBaselineState: editBaselineReducer,
        historicProfilesState: historicProfilesReducer,
        systemNotificationsState: systemNotificationsReducer,
        notifications,
        globalFilterState: globalFilterReducer
    });

    return registry;
}

export function getStore () {
    return registry.getStore();
}

export function register (...args) {
    return registry.register(...args);
}

export function addNewListener ({ actionType, callback }) {
    return middlewareListener.addNew({
        on: actionType,
        callback
    });
}
