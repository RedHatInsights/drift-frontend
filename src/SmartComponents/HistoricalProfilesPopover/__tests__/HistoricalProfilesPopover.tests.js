import React from 'react';
import { shallow, mount } from 'enzyme';
//import { MemoryRouter } from 'react-router-dom';
//import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';

//import api from '../../../api';
//import fixtures from './fixtures';
import { init } from '../../../store';
import { DropdownDirection } from '@patternfly/react-core';
import HistoricalProfilesPopover from '../HistoricalProfilesPopover';

jest.mock('../../../api');

/*eslint-disable camelcase*/
describe('HistoricalProfilesPopover', () => {
    let props;
    let { registry } = init();
    const store = registry.getStore();

    beforeEach(() => {
        props = {
            systemIds: [],
            selectedHSPIds: [],
            selectedBaselineIds: [],
            referenceId: undefined,
            dropdownDirection: DropdownDirection.down,
            hasBadge: true,
            hasCompareButton: false,
            badgeCount: 0,
            system: {
                id: '4416c520-a339-4dde-b303-f317ea9efc5f',
                last_updated: '2020-05-04T18:33:12.249348+00:00'
            },
            fetchCompare: jest.fn()
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <Provider store={ store }>
                <HistoricalProfilesPopover { ...props }/>
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mount correctly', () => {
        const wrapper = mount(
            <Provider store={ store }>
                <HistoricalProfilesPopover { ...props }/>
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it.skip('should call fetchCompare correctly', () => {
        props.systemIds = [ 'abc' ];
        props.selectedHSPIds = [ '123' ];
        props.selectedBaselineIds = [ 'def' ];
        props.referenceId = 'def';
        const wrapper = mount(
            <Provider store={ store }>
                <HistoricalProfilesPopover { ...props }/>
            </Provider>
        );

        wrapper.instance().fetchCompare();
        expect(props.fetchCompare).toHaveBeenCalledWith(
            props.systemIds, props.selectedBaselineIds, props.selectedHSPIds, props.referenceId
        );
    });

    describe('API', () => {
        it.skip('should set badgeCount to 0', () => {
            props.selectedHSPIds = [ 'abcd1234' ];
            props.badgeCount = 1;

            const wrapper = shallow(
                <Provider store={ store }>
                    <HistoricalProfilesPopover { ...props }/>
                </Provider>
            );

            wrapper.setState({
                historicalData: {
                    profiles: [{
                        captured_date: '2020-07-28T01:15:24.100078+00:00',
                        id: 'abcd1234',
                        system_id: '1234abcd'
                    }]
                }
            });
            expect(wrapper.state('badgeCount')).toBe(1);
            wrapper.setProps({ selectedHSPIds: []});
            wrapper.instance().updateBadgeCount();
            expect(wrapper.state('badgeCount')).toBe(0);
        });
    });
});

/*describe('ConnectedHistoricalProfilesPopover', () => {
    let initialState;
    let props;
    let mockStore;

    beforeEach(() => {
        mockStore = configureStore();
        initialState = {
            historicProfilesState: {
                selectedHSPIds: []
            }
        };

        props = {
            dropdownDirection: DropdownDirection.down,
            hasBadge: true,
            hasCompareButton: false,
            badgeCount: 0,
            system: {
                id: '4416c520-a339-4dde-b303-f317ea9efc5f',
                last_updated: '2020-05-04T18:33:12.249348+00:00'
            }
        };
    });

    it.skip('should render correctly', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedHistoricalProfilesPopover />
                </Provider>
            </MemoryRouter>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it.skip('should toggle dropdown menu', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedHistoricalProfilesPopover { ...props } />
                </Provider>
            </MemoryRouter>
        );

        wrapper.find('HistoryIcon').simulate('click');
        expect(wrapper.find('Popover').prop('isVisible')).toBe(true);
    });

    it.skip('should set historical data', () => {
        const store = mockStore(initialState);
        api.fetchHistoricalData = jest.fn();
        api.fetchHistoricalData
        .mockReturnValue(
            fixtures.historicalData
        );
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedHistoricalProfilesPopover
                        { ...props }
                    />
                </Provider>
            </MemoryRouter>
        );

        wrapper.find('HistoryIcon').simulate('click');
        expect(api.fetchHistoricalData).toHaveBeenCalled();
    });

    it.skip('should set error', async () => {
        const store = mockStore(initialState);
        api.fetchHistoricalData = jest.fn();
        api.fetchHistoricalData
        .mockReturnValue(
            fixtures.hspReturnedError
        );
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedHistoricalProfilesPopover
                        { ...props }
                    />
                </Provider>
            </MemoryRouter>
        );

        wrapper.find('HistoryIcon').simulate('click');
        wrapper.update();
        expect(api.fetchHistoricalData.mock.results[0].value.status).toBe(400);
        expect(api.fetchHistoricalData.mock.results[0].value.data.message).toBe('This is an error message');
    });

    it.skip('should call fetchCompare', async () => {
        const store = mockStore(initialState);
        props.hasCompareButton = true;
        const fetchCompare = jest.fn();
        api.fetchHistoricalData = jest.fn();
        api.fetchHistoricalData
        .mockReturnValue(
            fixtures.historicalData
        );
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedHistoricalProfilesPopover
                        { ...props }
                        fetchCompare={ fetchCompare }
                    />
                </Provider>
            </MemoryRouter>
        );

        wrapper.find('HistoryIcon').simulate('click');
        wrapper.update();
        wrapper.find('Button').simulate('click');
        expect(fetchCompare).toHaveBeenCalled();
    });

    it.skip('should retry fetch', async () => {
        const store = mockStore(initialState);
        api.fetchHistoricalData = jest.fn();
        api.fetchHistoricalData
        .mockReturnValue(
            fixtures.hspReturnedError
        );
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedHistoricalProfilesPopover
                        { ...props }
                    />
                </Provider>
            </MemoryRouter>
        );

        wrapper.find('HistoryIcon').simulate('click');
        wrapper.update();
        await wrapper.find('a').simulate('click');
        expect(api.fetchHistoricalData).toHaveBeenCalledTimes(2);
    });
});*/
/*eslint-enable camelcase*/
