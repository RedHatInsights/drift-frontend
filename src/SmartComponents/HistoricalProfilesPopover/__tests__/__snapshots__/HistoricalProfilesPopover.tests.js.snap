// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoricalProfilesPopover should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <HistoricalProfilesPopover
    badgeCount={0}
    dropdownDirection="down"
    fetchCompare={[MockFunction]}
    hasBadge={true}
    hasCompareButton={false}
    selectedBaselineIds={Array []}
    selectedHSPIds={Array []}
    system={
      Object {
        "id": "4416c520-a339-4dde-b303-f317ea9efc5f",
        "last_updated": "2020-05-04T18:33:12.249348+00:00",
      }
    }
    systemIds={Array []}
  />
</ContextProvider>
`;

exports[`HistoricalProfilesPopover should render mount correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HistoricalProfilesPopover
    badgeCount={0}
    dropdownDirection="down"
    fetchCompare={[MockFunction]}
    hasBadge={true}
    hasCompareButton={false}
    selectedBaselineIds={Array []}
    selectedHSPIds={Array []}
    system={
      Object {
        "id": "4416c520-a339-4dde-b303-f317ea9efc5f",
        "last_updated": "2020-05-04T18:33:12.249348+00:00",
      }
    }
    systemIds={Array []}
  >
    <span
      className="hsp-icon-padding"
      data-ouia-component-id="hsp-popover-toggle-4416c520-a339-4dde-b303-f317ea9efc5f"
      data-ouia-component-type="PF4/Button"
    >
      <Popover
        bodyContent={
          <div
            style={
              Object {
                "maxHeight": "350px",
                "overflowY": "scroll",
              }
            }
          >
            <React.Fragment>
              <Skeleton
                size="sm"
              />
              <br
                className="hsp-dropdown-loading"
              />
            </React.Fragment>
            <React.Fragment>
              <Skeleton
                size="sm"
              />
              <br
                className="hsp-dropdown-loading"
              />
            </React.Fragment>
            <React.Fragment>
              <Skeleton
                size="sm"
              />
              <br
                className="hsp-dropdown-loading"
              />
            </React.Fragment>
          </div>
        }
        footerContent={null}
        headerContent={
          <div>
            Historical profiles for this system
          </div>
        }
        id="hsp-popover-4416c520-a339-4dde-b303-f317ea9efc5f"
        isVisible={false}
        shouldClose={[Function]}
      >
        <Popper
          appendTo={[Function]}
          distance={25}
          enableFlip={true}
          flipBehavior={
            Array [
              "top",
              "bottom",
              "left",
              "right",
              "top-start",
              "top-end",
              "bottom-start",
              "bottom-end",
              "left-start",
              "left-end",
              "right-start",
              "right-end",
            ]
          }
          isVisible={false}
          onDocumentClick={[Function]}
          onDocumentKeyDown={[Function]}
          onTriggerClick={[Function]}
          placement="top"
          popper={
            <ForwardRef
              active={false}
              aria-describedby="popover-hsp-popover-4416c520-a339-4dde-b303-f317ea9efc5f-body"
              aria-labelledby="popover-hsp-popover-4416c520-a339-4dde-b303-f317ea9efc5f-header"
              aria-modal="true"
              className="pf-c-popover"
              focusTrapOptions={
                Object {
                  "clickOutsideDeactivates": true,
                  "fallbackFocus": [Function],
                  "returnFocusOnDeactivate": true,
                  "tabbableOptions": Object {
                    "displayCheck": "none",
                  },
                }
              }
              onMouseDown={[Function]}
              preventScrollOnDeactivate={true}
              role="dialog"
              style={
                Object {
                  "maxWidth": null,
                  "minWidth": null,
                  "opacity": 0,
                  "transition": "opacity 300ms cubic-bezier(.54, 1.5, .38, 1.11)",
                }
              }
            >
              <PopoverArrow />
              <PopoverContent>
                <PopoverCloseButton
                  aria-label="Close"
                  onClose={[Function]}
                />
                <PopoverHeader
                  alertSeverityScreenReaderText="undefined alert:"
                  icon={null}
                  id="popover-hsp-popover-4416c520-a339-4dde-b303-f317ea9efc5f-header"
                  titleHeadingLevel="h6"
                >
                  <div>
                    Historical profiles for this system
                  </div>
                </PopoverHeader>
                <PopoverBody
                  id="popover-hsp-popover-4416c520-a339-4dde-b303-f317ea9efc5f-body"
                >
                  <div
                    style={
                      Object {
                        "maxHeight": "350px",
                        "overflowY": "scroll",
                      }
                    }
                  >
                    <React.Fragment>
                      <Skeleton
                        size="sm"
                      />
                      <br
                        className="hsp-dropdown-loading"
                      />
                    </React.Fragment>
                    <React.Fragment>
                      <Skeleton
                        size="sm"
                      />
                      <br
                        className="hsp-dropdown-loading"
                      />
                    </React.Fragment>
                    <React.Fragment>
                      <Skeleton
                        size="sm"
                      />
                      <br
                        className="hsp-dropdown-loading"
                      />
                    </React.Fragment>
                  </div>
                </PopoverBody>
              </PopoverContent>
            </ForwardRef>
          }
          popperMatchesTriggerWidth={false}
          positionModifiers={
            Object {
              "bottom": "pf-m-bottom",
              "bottom-end": "pf-m-bottom-right",
              "bottom-start": "pf-m-bottom-left",
              "left": "pf-m-left",
              "left-end": "pf-m-left-bottom",
              "left-start": "pf-m-left-top",
              "right": "pf-m-right",
              "right-end": "pf-m-right-bottom",
              "right-start": "pf-m-right-top",
              "top": "pf-m-top",
              "top-end": "pf-m-top-right",
              "top-start": "pf-m-top-left",
            }
          }
          removeFindDomNode={false}
          trigger={
            <HistoryIcon
              className="hsp-dropdown-icon"
              color="currentColor"
              noVerticalAlign={false}
              onClick={[Function]}
              size="sm"
            />
          }
          zIndex={9999}
        >
          <FindRefWrapper
            onFoundRef={[Function]}
          >
            <HistoryIcon
              className="hsp-dropdown-icon"
              color="currentColor"
              noVerticalAlign={false}
              onClick={[Function]}
              size="sm"
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                className="hsp-dropdown-icon"
                fill="currentColor"
                height="1em"
                onClick={[Function]}
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M504 255.531c.253 136.64-111.18 248.372-247.82 248.468-59.015.042-113.223-20.53-155.822-54.911-11.077-8.94-11.905-25.541-1.839-35.607l11.267-11.267c8.609-8.609 22.353-9.551 31.891-1.984C173.062 425.135 212.781 440 256 440c101.705 0 184-82.311 184-184 0-101.705-82.311-184-184-184-48.814 0-93.149 18.969-126.068 49.932l50.754 50.754c10.08 10.08 2.941 27.314-11.313 27.314H24c-8.837 0-16-7.163-16-16V38.627c0-14.254 17.234-21.393 27.314-11.314l49.372 49.372C129.209 34.136 189.552 8 256 8c136.81 0 247.747 110.78 248 247.531zm-180.912 78.784l9.823-12.63c8.138-10.463 6.253-25.542-4.21-33.679L288 256.349V152c0-13.255-10.745-24-24-24h-16c-13.255 0-24 10.745-24 24v135.651l65.409 50.874c10.463 8.137 25.541 6.253 33.679-4.21z"
                />
              </svg>
            </HistoryIcon>
          </FindRefWrapper>
        </Popper>
      </Popover>
    </span>
  </HistoricalProfilesPopover>
</Provider>
`;
